### PR TITLE
fix: passing correct growerId to GrowerDetail

### LINKED ISSUE #998: Grower Detail: refactor to use new IDs only following migration
The `GrowerDetail` component still assumes we're working with the legacy API, which is no longer the case.

It needs a bit of refactoring to remove some of the code that handled both the legacy grower ID (now known as `grower.reference_id`) and new IDs (formerly known as `grower_account_uuid`, now just `grower.id`).

The component can be opened from various places, all of which have the new ID available to pass to the component:
1. Growers tool (`grower.id`)
2. Verify tool (`capture.grower_account_id`)
3. Capture Matching tool (`capture.grower_account_id`)
4. Earnings/Payments tool (`earnings.worker_id`)

### PR DESCRIPTION
**Pull request by pramodjsam**
## Description

Passing correct growerId to GrowerDetail page from different view ie Growers, Verify, Capture Matching Tool.

**Issue(s) addressed**

- Resolves #998 

**What kind of change(s) does this PR introduce?**

- [ ] Enhancement
- [x] Bug fix
- [ ] Refactor

**Please check if the PR fulfils these requirements**

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Review by nmcharlton — CHANGES_REQUESTED** [review]

File: src/components/GrowerDetail.js [hunk 1]
```diff
@@ -269,8 +269,12 @@ const GrowerDetail = ({ open, growerId, onClose }) => {
       }
     }
 
+    if (!grower && growerAccountUuid) {
+      grower = await api.getGrower(growerAccountUuid);
+    }
+
     if (!grower && !growerAccountUuid) {
```
[the remark this answers is not in the document]
Comment (pramodjsam) [reply src/components/GrowerDetail.js]: removed legacy api calling.

File: src/components/CaptureMatching/CaptureMatchingView.js [hunk 2]
```diff
@@ -331,7 +331,7 @@ function CaptureMatchingView() {
       }
     }
   }
-
+  console.log('GROWERACCOUNT', growerAccount);
```
[the remark this answers is not in the document]
Comment (pramodjsam) [reply src/components/CaptureMatching/CaptureMatchingView.js]: removed the console.log's

File: src/components/GrowerDetail.js [hunk 3]
```diff
@@ -269,8 +269,12 @@ const GrowerDetail = ({ open, growerId, onClose }) => {
       }
     }
```
[the remark this answers is not in the document]
Comment (pramodjsam) [reply src/components/GrowerDetail.js:270]: removed the code block for the legacy api call.

image

[the remark this answers is not in the document]
Comment (nmcharlton) [reply src/components/GrowerDetail.js]: I've pushed a commit which removes the rest of the now unused code. We no longer need to cater for two types of ID.

**Review by nmcharlton — APPROVED** [review]

**Pull request merged**